gads.py

```diff
@@ -17,41 +17,48 @@
 from flask import session
 from google.ads.googleads.client import GoogleAdsClient
 from fs_storage import get_value_session
+import logging
 
 
 def get_gads_client(mcc_id: str) -> GoogleAdsClient:
-  main_creds = json.loads(get_value_session(session['session_id'], 'credentials'))
-  creds = {
-      "developer_token": os.environ["GOOGLE_ADS_DEVELOPER_TOKEN"],
-      "refresh_token": main_creds["refresh_token"],
-      "client_id": main_creds["client_id"],
-      "client_secret": main_creds["client_secret"],
-      "use_proto_plus": True,
-  }
-
-  google_ads_client = GoogleAdsClient.load_from_dict(creds)
-  if mcc_id:
-    google_ads_client.login_customer_id = mcc_id
-
-  return google_ads_client
+  try:
+    main_creds = json.loads(get_value_session(session['session_id'], 'credentials'))
+    creds = {
+        "developer_token": os.environ["GOOGLE_ADS_DEVELOPER_TOKEN"],
+        "refresh_token": main_creds["refresh_token"],
+        "client_id": main_creds["client_id"],
+        "client_secret": main_creds["client_secret"],
+        "use_proto_plus": True,
+    }
+
+    google_ads_client = GoogleAdsClient.load_from_dict(creds)
+    if mcc_id:
+        google_ads_client.login_customer_id = mcc_id
+
+    return google_ads_client
+  except ValueError as e:
+    logging.exception(e)
+    return f"Error: Session Expired ({e})"
 
 
 def get_gads_data(mcc_id: str, customer_id: str, campaign_ids: list, date_from: str, date_to: str) -> dict:
   client = get_gads_client(mcc_id)
-  ga_service = client.get_service("GoogleAdsService", version="v14")
-  camp_filter = ""
-  if not customer_id + "-0" in campaign_ids[0]:
-    camp_filter = "AND campaign.id IN (" + (",").join(campaign_ids) + ")"
-
-  query = textwrap.dedent(f"""
-    SELECT segments.date, metrics.impressions, metrics.clicks, metrics.video_views, metrics.cost_micros, metrics.conversions, metrics.view_through_conversions FROM campaign
-    WHERE segments.date BETWEEN '{date_from}' AND '{date_to}' {camp_filter}
-    ORDER BY segments.date ASC
-    """)
-
-  response = ga_service.search_stream(customer_id=customer_id, query=query)
-  response.service_reference = ga_service
-  return response
+  if "Error" not in client:
```

Comment:
No need to change but similar to earlier, I think this reads better as

```
if "Error" in client:
    return "Session Expired"
```

Reply:
Fixed